src/main/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/config/SelfDestructConfig.kt

```diff
@@ -0,0 +1,57 @@
+package no.nav.soknad.arkivering.soknadsarkiverer.config
+
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+
+@Configuration
+class SelfDestructConfig(private val scheduler: Scheduler, private val appConfiguration: AppConfiguration) {
+	private val logger = LoggerFactory.getLogger(javaClass)
+
+	/**
+	 * The application runs in many different pods, as configured in .nais/nais.yml, where the standard configuration is
```

Comment:
Jeg har et forslag til beskrivelse av feilen som kan kunne oppstå ved en nedskalering av podder. Du kan kanskje kombinere den med din beskrivelse for å knytte den nærmere implementasjonen?

The application subscribe to soknad_innsending to topic. This topic is configured to have 4 partitions (lets call them P1, P2, P3 and P4). The Kafka broker will automatically put a new message to one of the partitions, and allocate each partitions to one of the running pods of the application. This means that one pod will process all messages on one partition. If the number of running pods changes, the kafka broker will automatically run re-partitioning.  Example: the number of running pods is 3 and pod 1 handels partition P1 and P2,  pod 2 handels P3 and pod 3 handels P4. If kubernetes decides to downscale to 2 pods, the Kafka broker will automatically re-assign the partitions between the remainding pods. Pod 1 handels partition P1 and P2 and P3 and P4. This means that pod 2 will start processing **new** messages on P4 (that where previously assigned to killed pod 3). However, messages from P4 that pod 3 where processing when when it was killed, will not be picked up. In order to ensure that these are picked up the pods needs to be re-started because during re-start the soknad_innsending topic is re-read for processing all unfinished task.

Reply:
Takk, meget bra, skal slå sammen teksterne.